doc/connectivity-semantics-and-api.md

```diff
@@ -75,6 +90,15 @@ corresponding reasons. Empty cells denote disallowed transitions.
     <td></td>
     <td></td>
     <td></td>
+    <td>Shutdown triggered by application.</td>
+  </tr>
+  <tr>
+    <th>IDLE</th>
+    <td>Any new RPC activity on the channel</td>
```

Comment:
Then what state would we be in? I'm not sure I understand this document then.

If we get a GOAWAY then we have to reconnect for new RPCs and go into CONNECTING. That reconnect could fail and cause us to be in TRANSIENT_FAILURE. Would we not go through those states?

If we don't go to IDLE, then how do we go to CONNECTING? Go directly there from READY, without going through IDLE/TRANSIENT_FAILURE?